### PR TITLE
Issue #1026

### PR DESCRIPTION
**Pull request by beercsaba**
implementation of Issue #1026

### CODE REVIEW

File: Source/LinqToDB/DataExtensions.cs [hunk 1]
```diff
@@ -462,12 +503,12 @@ public static int Delete<T>([NotNull] this IDataContext dataContext, T obj)
 		/// </param>
 		/// <returns>Created table as queryable source.</returns>
 		public static ITable<T> CreateTable<T>([NotNull] this IDataContext dataContext,
-			string          tableName       = null,
-			string          databaseName    = null,
-			string          schemaName      = null,
-			string          statementHeader = null,
-			string          statementFooter = null,
-			DefaultNullable defaultNullable  = DefaultNullable.None)
+			string tableName = null,
```
Comment (sdanyliv) [inline Source/LinqToDB/DataExtensions.cs]: Please return table formatting back. This comment is related to whole file.

Comment (beercsaba) [reply Source/LinqToDB/DataExtensions.cs]: I saw this issue. my VS made this formatting :(
Do you know any way to set the tabular formatting in VS?

Comment (sdanyliv) [reply Source/LinqToDB/DataExtensions.cs]: We just undo VS fromatting. And trying to disable, formatting on paste, after semicolon.
Igor made this formatting by hands to better read and understand code.

Comment (sdanyliv) [reply Source/LinqToDB/DataExtensions.cs]: Before commit we ensure that we do not change non-related parts.

File: Source/LinqToDB/Linq/QueryRunner.Insert.cs [hunk 2]
```diff
@@ -1,5 +1,4 @@
-﻿using System;
-using System.Collections.Concurrent;
+﻿using System.Collections.Concurrent;
```
Comment (sdanyliv) [inline Source/LinqToDB/Linq/QueryRunner.Insert.cs]: Looks like this file do not need changes

File: Source/LinqToDB/Linq/QueryRunner.InsertOrReplace.cs [hunk 3]
```diff
@@ -14,18 +13,23 @@ static partial class QueryRunner
...
-			static readonly ConcurrentDictionary<object,Query<int>> _queryChache = new ConcurrentDictionary<object,Query<int>>();
+			static readonly ConcurrentDictionary<object, Query<int>> _queryCache = new ConcurrentDictionary<object, Query<int>>();
 
-			static Query<int> CreateQuery(IDataContext dataContext)
+			static Query<int> CreateQuery(IDataContext dataContext, string tableName = null, string databaseName = null, string schemaName = null)
 			{
 				var fieldDic = new Dictionary<SqlField,ParameterAccessor>();
 				var sqlTable = new SqlTable<T>(dataContext.MappingSchema);
+
+				if (tableName		!= null)	sqlTable.PhysicalName	= tableName;
```
Comment (sdanyliv) [inline Source/LinqToDB/Linq/QueryRunner.InsertOrReplace.cs]: Fromatting rules, are: line starts from Tab, then formatting by spaces. You can see online that formatting is broken.

**Comment by sdanyliv** [comment]
If you doing rollbacks by hands, do’t do that i will fix everything tomorrow.
Next time be careful with such kind of commits.
Thanks!

File: Source/LinqToDB/Linq/QueryRunner.Insert.cs (hunk 2 above)
Comment (beercsaba) [reply Source/LinqToDB/Linq/QueryRunner.Insert.cs]: i've fixed a typo (in these QueryRunner.* classes): _queryChache should be spelled queryCache

**Comment by beercsaba** [comment]
Yes, i'm doing it by hands.  I don't want to break your rules.
There are some small differences remained, but maybe (i hope) they are acceptable.

imho: 
from some point of view It would be better to wrap this Collection<Entity ->  (tableName, schemaName, databaseName) triplet> mapping into a CustomNaming class and make it a field of the DataContext class, which can be set in constructor. This would restrict to one-to-one mapping, but make method calls shorter (no need of these 3 parameters), so code will be easier to read. 
What do you think?
I have no time for working on this now and the next 2 months, so if you agree with my opinion, i could work on that later.

File: Source/LinqToDB/DataExtensions.cs [hunk 4]
```diff
@@ -49,10 +47,10 @@ public static ITable<T> GetTable<T>(this IDataContext dataContext)
 		[LinqTunnel]
 		[Pure]
 		public static ITable<T> GetTable<T>(
-			this IDataContext dataContext,
-			object instance,
-			[NotNull] MethodInfo methodInfo,
-			[NotNull] params object[] parameters)
+						this IDataContext	dataContext,
```
Comment (sdanyliv) [inline Source/LinqToDB/DataExtensions.cs]: Undo formatting. Leave changed only your methods.

Comment (beercsaba) [reply Source/LinqToDB/DataExtensions.cs]: done

File: Source/LinqToDB/DataExtensions.cs [hunk 5]
```diff
@@ -98,9 +96,9 @@ public static ITable<T> GetTable<T>(this IDataContext dataContext)
 		/// <param name="query">The query expression to be compiled.</param>
 		/// <typeparam name="TDc">Type of data context parameter, passed to compiled query.</typeparam>
 		/// <typeparam name="TResult">Query result type.</typeparam>
-		public static Func<TDc,TResult> Compile<TDc,TResult>(
+		public static Func<TDc, TResult> Compile<TDc, TResult>(
```
Comment (sdanyliv) [inline Source/LinqToDB/DataExtensions.cs]: Spaces also revert, its’ specific Igor’s style, no spaces between generic types.
For me is better to revert whole file and just copy paste your changes from previous version.

**Comment by beercsaba** [comment]
I think i'm done with formatting, which takes almost as much time as the implementation.
i think i've fixed some formatting inconsistencies in these files. If you do not agree, i could revert these fixes.
i've renamed field _queryChache to _queryCache in there QueryRunner.... classes, i think this is a typo. If you do not agree, i could revert these too.
@sdanyliv what do you think?

**Comment by sdanyliv** [comment]
@beercsaba, i'm picking your PR and will made corrections. Thanks for your help.

**Comment by sdanyliv** [comment]
I have opened new PR #1045.

**Pull request closed without merging**